src/hal/watchdog.cpp

```diff
@@ -0,0 +1,66 @@
+
+#include "../hal/hal.h"
+#include "../hal/watchdog.h"
+
+#if defined(__AVR__)
+  #include <avr/wdt.h>
+#elif defined(__arm__)
+// TODO: Consider making this more specific if needed
+  #include <include/wdt.h>
+#else
+  #error "Unsupported Architecture for Watchdog HAL"
+#endif
+
+#if defined(__arm__)
+// Atmel SAM3X8E only allows a single write to WDT_MR; which is written to by
+// the core libraries to disable the watchdog by default; override this function
+// to prevent that
+void watchdogSetup(void) {
+  return;
+}
+#endif
+
+int watchdogHalInit(void)
+{
+#if defined(__AVR__)
+  if (WATCHDOG_PERIOD_MS == 15) {
```

Comment:
Maybe replace these with a `switch` statement?

Reply:
I had a reason I didnt use a `switch` here, but now I dont remember what it was; so yeah, lets use a `switch`

Reply:
Done in latest commits